### PR TITLE
Eye damage

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
Adds eye damage, mostly from welding right now. Also adds welding masks for protection, glasses to correct eye damage, oculine as a chem to heal it, and carrot juice now metabolizes into oculine.

Closes https://github.com/space-wizards/space-station-14/issues/3276

**FAQ**
Q: Doesn't this need bobby?
Shrimply map the damage to the eye organ to the 8 current thresholds after bobby

**Changelog**

:cl: Rane
- add: Welding without eye protection now blinds you, and causes permanent eye damage if you had no protection at all.
- add: Added oculine to heal eye damage.
- add: Added glasses to correct blurry vision.
- tweak: Renamed mesons to engineering goggles for less confusion. They no longer provide radiation damage, instead functioning more as a compact welding mask.
- fix: You will no longer see 'this entity does not exist' when examining entities that exist.

### CODE REVIEW

**Comment by lapatison** [comment]
Probably, related to #3276

**Comment by Elijahrane** [comment]
> Probably, related to #3276

Should outright close that probably.

Also it's been like 5 days now without reported eye damage bugs downstream

File: Content.Client/Eye/Blinding/BlurryVisionOverlay.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        public override OverlaySpace Space => OverlaySpace.WorldSpace;
+        private readonly ShaderInstance _blurryVisionXShader;
+        private readonly ShaderInstance _blurryVisionYShader;
+        private BlurryVisionComponent _blurryVisionComponent = default!;
+
+        public BlurryVisionOverlay()
+        {
+            IoCManager.InjectDependencies(this);
+            _blurryVisionXShader = _prototypeManager.Index<ShaderPrototype>("BlurryVisionX").InstanceUnique();
+            _blurryVisionYShader = _prototypeManager.Index<ShaderPrototype>("BlurryVisionY").InstanceUnique();
```
Comment (mirrorcult) [inline Content.Client/Eye/Blinding/BlurryVisionOverlay.cs:26]: why do these have to be two different shaders?

File: Content.Client/Eye/Blinding/BlurryVisionSystem.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+
```
Comment (mirrorcult) [inline Content.Client/Eye/Blinding/BlurryVisionSystem.cs:1]:
```suggestion
```

File: Content.Client/Eye/Blinding/BlurryVisionOverlay.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+using Content.Shared.Eye.Blinding;
+
+namespace Content.Client.Eye.Blinding
+{
+    public sealed class BlurryVisionOverlay : Overlay
+    {
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] IEntityManager _entityManager = default!;
+        [Dependency] private readonly IPlayerManager _playerManager = default!;
+
```
Comment (mirrorcult) [inline Content.Client/Eye/Blinding/BlurryVisionOverlay.cs:14]:
```suggestion
```

File: Content.Client/Eye/Blinding/BlurryVisionOverlay.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+using Robust.Shared.Enums;
+using Robust.Shared.Prototypes;
+using Content.Shared.Eye.Blinding;
+
+namespace Content.Client.Eye.Blinding
+{
+    public sealed class BlurryVisionOverlay : Overlay
+    {
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] IEntityManager _entityManager = default!;
```
Comment (mirrorcult) [inline Content.Client/Eye/Blinding/BlurryVisionOverlay.cs:12]:
```suggestion
        [Dependency] private readonly IEntityManager _entityManager = default!;
```

File: Content.Server/Eye/Blinding/EyeProtection/EyeProtectionSystem.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                return;
+
+            float statusTime = (float) component.StatusEffectTime.TotalSeconds - blindable.BlindResistance;
+
+            if (statusTime <= 0)
+                return;
+
+            var statusTimeSpan = TimeSpan.FromSeconds(statusTime * (blindable.EyeDamage + 1));
+            // Add permanent eye damage if they had zero protection, also scale their temporary blindness by how much they already accumulated.
+            if (_statusEffectsSystem.TryAddStatusEffect(args.User, "TemporaryBlindness", statusTimeSpan, false, "TemporaryBlindness") && blindable.BlindResistance <= 0)
```
Comment (mirrorcult) [inline Content.Server/Eye/Blinding/EyeProtection/EyeProtectionSystem.cs:43]: make the key a static string

File: Content.Shared/Examine/ExamineSystemShared.cs [hunk 6]
```diff
@@ -101,6 +101,11 @@ public float GetExaminerRange(EntityUid examiner, MobStateComponent? mobState =
                     return DeadExamineRange;
                 else if (MobStateSystem.IsCritical(examiner, mobState) || (TryComp<BlindableComponent>(examiner, out var blind) && blind.Sources > 0))
                     return CritExamineRange;
+                else if (TryComp<BlurryVisionComponent>(examiner, out var blurry) && blurry.Magnitude != 0)
```
Comment (mirrorcult) [inline Content.Shared/Examine/ExamineSystemShared.cs:104]:
```suggestion

                else if (TryComp<BlurryVisionComponent>(examiner, out var blurry) && blurry.Magnitude != 0)
```

File: Resources/Prototypes/Entities/Clothing/Eyes/glasses.yml [hunk 7]
```diff
@@ -65,17 +65,15 @@
...
   components:
   - type: Sprite
     sprite: Clothing/Eyes/Glasses/meson.rsi
   - type: Clothing
     sprite: Clothing/Eyes/Glasses/meson.rsi
-  - type: Armor
-    modifiers:
-      coefficients:
-        Radiation: 0.5
+  - type: FlashImmunity
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Clothing/Eyes/glasses.yml:75]: giving these flash immunity is potentially a bit much with revs

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Client/Eye/Blinding/BlurryVisionOverlay.cs (hunk 1 above)
Comment (Elijahrane) [reply Content.Client/Eye/Blinding/BlurryVisionOverlay.cs:26]: I'm not a shader guy I just copied from godot documentation

Comment (Elijahrane) [reply Content.Client/Eye/Blinding/BlurryVisionOverlay.cs:26]: All of the more obvious ways to do it do not work on our rendering engine (at least not for older machines)

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**